File: e2e-test/local-storage-module/android/src/main/java/com/modules/localstorage/NativeLocalStorageModule.java
package com.modules.localstorage;

import android.content.Context;
import android.content.SharedPreferences;
import com.modules.localstorage.generated.NativeLocalStorageModuleSpec;
import com.lynx.react.bridge.Callback;
import com.tigermodule.autolink.LynxNativeModule;

/**
 * Implementation of NativeLocalStorageModule
 * Extend the generated base class and implement your logic
 */
@LynxNativeModule(name = "NativeLocalStorageModule")
public class NativeLocalStorageModule extends NativeLocalStorageModuleSpec {
  private static final String PREF_NAME = "MyLocalStorage";
  
  public NativeLocalStorageModule(Context context) {
    super(context);
  }

  @Override
  public void setStorageItem(String key, String value) {
    SharedPreferences sharedPreferences = getContext().getSharedPreferences(PREF_NAME, Context.MODE_PRIVATE);
    SharedPreferences.Editor editor = sharedPreferences.edit();
    editor.putString(key, value);
    editor.apply();
  }

  @Override
  public void getStorageItem(String key, Callback callback) {
    SharedPreferences sharedPreferences = getContext().getSharedPreferences(PREF_NAME, Context.MODE_PRIVATE);
    String value = sharedPreferences.getString(key, null);
    callback.invoke(value);
  }

  @Override
  public void clearStorage() {
    SharedPreferences sharedPreferences = getContext().getSharedPreferences(PREF_NAME, Context.MODE_PRIVATE);
    SharedPreferences.Editor editor = sharedPreferences.edit();
    editor.clear();
    editor.apply();
  }
}

File: e2e-test/module-element-testing/android/src/main/kotlin/com/testing/NativeLocalStorageModule.kt
package com.testing

import android.content.Context
import com.testing.generated.NativeLocalStorageModuleSpec
import com.lynx.react.bridge.Callback
import com.tigermodule.processor.LynxNativeModule
import com.lynx.tasm.behavior.LynxContext

/**
 * Implementation of NativeLocalStorageModule
 * Extend the generated base class and implement your logic
 */
@LynxNativeModule(name = "NativeLocalStorageModule")
class NativeLocalStorageModule(context: Context) : NativeLocalStorageModuleSpec(context) {

  private val PREF_NAME = "MyLocalStorage"
  
  override fun setStorageItem(key: String, value: String) {
    val sharedPreferences = getContext().getSharedPreferences(PREF_NAME, Context.MODE_PRIVATE)
    val editor = sharedPreferences.edit()
    editor.putString(key, value)
    editor.apply()
  }

  override fun getStorageItem(key: String, callback: Callback) {
    val sharedPreferences = getContext().getSharedPreferences(PREF_NAME, Context.MODE_PRIVATE)
    callback.invoke(sharedPreferences.getString(key, null))
  }

  override fun clearStorage() {
    val sharedPreferences = getContext().getSharedPreferences(PREF_NAME, Context.MODE_PRIVATE)
    val editor = sharedPreferences.edit()
    editor.clear()
    editor.apply()
  }
}